history/taskana-simplehistory-provider/src/main/java/pro/taskana/simplehistory/impl/HistoryCleanupJob.java

```diff
@@ -0,0 +1,337 @@
+import java.time.Duration;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import pro.taskana.TaskanaEngineConfiguration;
+import pro.taskana.common.api.ScheduledJob;
+import pro.taskana.common.api.ScheduledJob.Type;
+import pro.taskana.common.api.TaskanaEngine;
+import pro.taskana.common.api.TimeInterval;
+import pro.taskana.common.api.exceptions.InvalidArgumentException;
+import pro.taskana.common.api.exceptions.NotAuthorizedException;
+import pro.taskana.common.api.exceptions.SystemException;
+import pro.taskana.common.api.exceptions.TaskanaException;
+import pro.taskana.common.internal.jobs.AbstractTaskanaJob;
+import pro.taskana.common.internal.transaction.TaskanaTransactionProvider;
+
+public class HistoryCleanupJob extends AbstractTaskanaJob {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(HistoryCleanupJob.class);
+
+  private static final String TASKANA_PROPERTIES = "/taskana.properties";
+
+  private static final String TASKANA_JOB_HISTORY_BATCH_SIZE = "taskana.jobs.history.batchSize";
+  private static final String TASKANA_JOB_HISTORY_CLEANUP_RUN_EVERY =
+      "taskana.jobs.history.cleanup.runEvery";
+  private static final String TASKANA_JOB_HISTORY_CLEANUP_FIRST_RUN =
+      "taskana.jobs.history.cleanup.firstRunAt";
+  private static final String TASKANA_JOB_HISTORY_CLEANUP_MINIMUM_AGE =
+      "taskana.jobs.history.cleanup.minimumAge";
+
+  TaskanaHistoryEngineImpl taskanaHistoryEngine =
+      TaskanaHistoryEngineImpl.createTaskanaEngine(taskanaEngineImpl.getConfiguration());
+
+  private Instant firstRun;
+  private Duration runEvery;
+  private Duration minimumAge;
+  private int batchSize;
+  private boolean allCompletedSameParentBusiness;
+
+  public HistoryCleanupJob(
+      TaskanaEngine taskanaEngine,
+      TaskanaTransactionProvider<Object> txProvider,
+      ScheduledJob scheduledJob) {
+    super(taskanaEngine, txProvider, scheduledJob);
+    allCompletedSameParentBusiness =
+        taskanaEngine.getConfiguration().isTaskCleanupJobAllCompletedSameParentBusiness();
+    Properties props = readPropertiesFromFile(TASKANA_PROPERTIES);
+    initJobParameters(props);
+  }
+
+  @Override
+  public void run() throws TaskanaException {
+
+    Instant createdBefore = Instant.now().minus(minimumAge);
+
+    LOGGER.info("Running job to delete all history events created before ({})", createdBefore);
+
+    try {
+
+      SimpleHistoryServiceImpl simpleHistoryService =
+          (SimpleHistoryServiceImpl) taskanaHistoryEngine.getTaskanaHistoryService();
+
+      List<HistoryEventImpl> historyEventsToClean =
+          simpleHistoryService
+              .createHistoryQuery()
+              .createdWithin(new TimeInterval(null, createdBefore))
+              .eventTypeIn("TASK_COMPLETED")
```

Comment:
I don't think that this is correct. We do want to delete all events for the tasks and not just the completed event.